### PR TITLE
tuna/tables_interface.py - type hint support

### PR DESCRIPTION
**Pull request by technicalgrp89**
mypy compatible code
type hints support

### CODE REVIEW

File: tuna/tables_interface.py [hunk 1]
```diff
@@ -26,32 +26,35 @@
...
+    self.job_table: Optional[Dict[str, Any]] = None
+    self.session_id: Optional[int] = None
+    self.session: Query = None
 
     self.__dict__.update(
         (key, value) for key, value in kwargs.items() if key in allowed_keys)
 
-  def set_tables(self, sess_class):
+  def do_nothing(self):
+    """dummy function to avoid R0903"""
```
Comment (alexandraBara) [inline tuna/tables_interface.py]: remove and add the pylint ignore if needed

File: tuna/tables_interface.py [hunk 2]
```diff
@@ -26,32 +26,33 @@
...
     super().__init__()
-    allowed_keys = set(['session_id'])
+    allowed_keys: Set = set(['session_id'])
     self.__dict__.update((key, None) for key in allowed_keys)
 
-    #for pylint
-    self.job_table = None
-    self.session_id = None
-    self.session = None
+    self.job_table: Optional[Dict[str, Any]] = None
```
Comment (alexandraBara) [inline tuna/tables_interface.py]: this is not a dict, this should be an option between: ConvolutionJob, BNJob and FusionJob class defined in /miopen/db/miopen_tables.py. Basically any of the classes that implement the JobMixin Interface

File: tuna/tables_interface.py [hunk 3]
```diff
@@ -26,32 +26,33 @@
...
+    self.session_id: Optional[int] = None
+    self.session: Query = None
```
Comment (alexandraBara) [inline tuna/tables_interface.py]: this is not a Query its any of the classes that implements the SessionMixin interface, i think they might be ExampleSession and Session from MIOpen or something like that, you would have to look it up.

**Review by alexandraBara — CHANGES_REQUESTED** [review]

File: tuna/tables_interface.py [hunk 4]
```diff
@@ -26,28 +26,33 @@
...
     super().__init__()
-    allowed_keys = set(['session_id'])
+    allowed_keys: Set = set(['session_id'])
     self.__dict__.update((key, None) for key in allowed_keys)
 
-    #for pylint
-    self.job_table = None
-    self.session_id = None
-    self.session = None
+    self.job_table: Optional[Union[ConvolutionJob, BNJob, FusionJob,
```
Comment (cderb) [inline tuna/tables_interface.py]: Simplify this to the parent class. These are all JobMixin.

File: tuna/tables_interface.py [hunk 5]
```diff
@@ -26,28 +26,33 @@
...
+                                   Job]] = None
+    self.session_id: Optional[int] = None
+    self.session: Optional[Union[Session, SessionExample]] = None
```
Comment (cderb) [inline tuna/tables_interface.py]: Use base class SessionMixin.

**Review by cderb — CHANGES_REQUESTED** [review]

File: tuna/tables_interface.py [hunk 6]
```diff
@@ -26,28 +26,30 @@
...
+from tuna.session_mixin import SessionMixin
+from tuna.miopen.db.miopen_tables import JobMixin
 
 
 #pylint: disable=too-few-public-methods
 class DBTablesInterface():
   """Represents db tables interface class"""
 
-  def __init__(self, **kwargs):
+  def __init__(self, **kwargs: Dict[str, Dict[str, Any]]):
```
Comment (cderb) [inline tuna/tables_interface.py:39]: @alexandraBara do you know the structure of kwargs here?

**Review by cderb — APPROVED** [review]

File: tuna/tables_interface.py (hunk 6 above)
Comment (alexandraBara) [reply tuna/tables_interface.py:39]: looks like currently its just a dict that takes in session_id as a key. I think a dict of str to Any should be comprehensive for future

File: tuna/tables_interface.py (hunk 4 above)
Comment (alexandraBara) [reply tuna/tables_interface.py]: this is nice, didnt realize you can do this, simplifies things

**Review by alexandraBara — APPROVED** [review]

**Review by JehandadKhan — APPROVED** [review]

**Pull request merged**